components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/services/TenantMgtAdminService.java

```diff
@@ -572,38 +575,71 @@ public void deactivateTenant(String tenantDomain) throws Exception {
      * @param tenantDomain The domain name of the tenant that needs to be deleted
      */
     public void deleteTenant(String tenantDomain) throws StratosException, org.wso2.carbon.user.api.UserStoreException {
+
         TenantManager tenantManager = TenantMgtServiceComponent.getTenantManager();
         if (tenantManager != null) {
             int tenantId = tenantManager.getTenantId(tenantDomain);
             try {
                 log.info("Starting Tenant Deletion process...");
 
                 notifyTenantDeletion(tenantId);
+                // Before deleting the tenant deactivate the tenant from the system.
+                // Deactivation also used to clear the tenant configuration context for that particular tenant.
+                deactivateTenant(tenantDomain);
```

Comment:
Shouldn't the tenant deactivation suppose to happen after checking the tenant deletion property is set to true?

Reply:
Yes.ACK